synapse/util/caches/dictionary_cache.py

```diff
@@ -149,20 +286,27 @@ def update(
             # Only update the cache if the caches sequence number matches the
             # number that the cache had before the SELECT was started (SYN-369)
             if fetched_keys is None:
-                self._insert(key, value, set())
+                self.cache[(key, _FullCacheKey.KEY)] = value
```

Comment:
do we not need to update (or at least invalidate) entries for individual dict keys here as well?

Reply:
I don't think so, if the value in the DB has changed then we should have previously called `invalidate(..)` to delete the stale entries? We could easily drop any existing entries, but I don't think its necessary?

Reply:
fair enough. That seems like an assumption that we should document in the docstring though.

(Also: shouldn't this method be called `set` or `put` or `insert` or something? Maybe a problem for another time though.)